### PR TITLE
Mention the need for Subscription Reader role

### PR DESCRIPTION
**Pull request by lupino3**
The app registration will need to have the Subscription Reader role to be able to enumerate resources and do other operations.

I discovered this together with the Grafana support, as I was playing with Grafana Cloud using resources in my personal Azure subscription.

**Special notes for your reviewer**: I verified this with Andrew T. from the Grafana Support team.

### CODE REVIEW

File: docs/sources/datasources/azuremonitor/_index.md [hunk 1]
```diff
@@ -31,7 +31,9 @@ This topic explains configuring, querying, and other options specific to the Azu
...
+You must create an app registration and service principal in Azure AD to authenticate the data source. See the [Azure documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in) for configuration details. The app registration you create will need to have the `Reader` role assigned on the subscription. See the [Azure docs for role assignments](https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal?tabs=current) for more information.
+
+Alternatively, if you are hosting Grafana in Azure (e.g. App Service, or Azure Virtual Machines) you can configure the Azure Monitor data source to use Managed Identity to securely authenticate without entering credentials into Grafana. Refer to [Configuring using Managed Identity](#configuring-using-managed-identity) for more details.
```
Comment (brendamuir) [inline docs/sources/datasources/azuremonitor/_index.md:36]:
```suggestion
You must create an app registration and service principal in Azure AD to authenticate the data source. For more information, refer to [Azure documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in) for configuration details. The app registration you create will need to have the `Reader` role assigned on the subscription. For more information, refer to [Azure docs for role assignments](https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal?tabs=current).

Alternatively, if you are hosting Grafana in Azure (e.g. App Service, or Azure Virtual Machines) you can configure the Azure Monitor data source to use Managed Identity to securely authenticate without entering credentials into Grafana. Refer to [Configuring using Managed Identity](#configuring-using-managed-identity) for more details.
```

File: docs/sources/datasources/azuremonitor/_index.md [hunk 2]
```diff
@@ -31,7 +31,9 @@ This topic explains configuring, querying, and other options specific to the Azu
...
+You must create an app registration and service principal in Azure AD to authenticate the data source. For more information, refer to [Azure documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in) for configuration details. The app registration you create will need to have the `Reader` role assigned on the subscription. For more information, refer to [Azure docs for role assignments](https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal?tabs=current).
```
Comment (andresmgot) [inline docs/sources/datasources/azuremonitor/_index.md:34]: hi @lupino3 I believe that the minimal set of permissions necessary is read access to a resource group (not necessarily an entire subscription). Can you check if you are able to use the plugin limiting the access to one (or more) a resource group(s)?

**Review by brendamuir — APPROVED** [review]

File: docs/sources/datasources/azuremonitor/_index.md (hunk 2 above)
Comment (lupino3) [reply docs/sources/datasources/azuremonitor/_index.md:34]: I was trying to do it, but got some unrelated errors and have started talking to support again. Will do once I understand if there are other problems.

File: docs/sources/datasources/azuremonitor/_index.md [hunk 3]
```diff
@@ -31,7 +31,9 @@ This topic explains configuring, querying, and other options specific to the Azu
...
+
+Alternatively, if you are hosting Grafana in Azure (e.g. App Service, or Azure Virtual Machines) you can configure the Azure Monitor data source to use Managed Identity to securely authenticate without entering credentials into Grafana. Refer to [Configuring using Managed Identity](#configuring-using-managed-identity) for more details.
```
Comment (gguillotte-grafana) [inline docs/sources/datasources/azuremonitor/_index.md:36]: Small style/phrasing suggestion:

```suggestion
Alternatively, if you host Grafana in Azure (such as in App Service or Azure Virtual Machines), you can configure the Azure Monitor data source to use Managed Identity to securely authenticate without entering credentials into Grafana. Refer to [Configuring using Managed Identity](#configuring-using-managed-identity) for details.
```

**Review by gguillotte-grafana — APPROVED** [review]
Made a small non-blocking suggestion to adhere to docs style guides.

File: docs/sources/datasources/azuremonitor/_index.md (hunk 2 above)
Comment (andresmgot) [reply docs/sources/datasources/azuremonitor/_index.md:34]: Lets merge this as it is, in any case having permissions for a subscription seems like the right way to go.

**Comment by lupino3** [comment]
Apologies, but I had a pretty busy time and couldn't get back to playing with Grafana.

**Comment by andresmgot** [comment]
hi @lupino3 do you mind merging the latest `main` changes here so the CI check passes?

**Comment by lupino3** [comment]
> hi @lupino3 do you mind merging the latest `main` changes here so the CI check passes?

I just did it, let's see if robots are happy now. Thanks! :)

**Comment by andresmgot** [comment]
Thanks!

**Pull request merged**